jedi/refactoring.py

```diff
@@ -12,86 +12,146 @@
 - extract variable
 - inline variable
 """
-import difflib
+from difflib import unified_diff
+from collections import namedtuple
 
+from itertools import groupby
 from jedi import common
 from jedi.evaluate import helpers
 from jedi.parser import tree as pt
+from common import content, source_to_unicode, splitlines
 
 
-class Refactoring(object):
-    def __init__(self, change_dct):
-        """
-        :param change_dct: dict(old_path=(new_path, old_lines, new_lines))
-        """
-        self.change_dct = change_dct
+class Pos(namedtuple('Position', ['line', 'column'])):
+    @property
+    def real_line(self):
+        return self.line - 1
 
-    def old_files(self):
-        dct = {}
-        for old_path, (new_path, old_l, new_l) in self.change_dct.items():
-            dct[new_path] = '\n'.join(new_l)
-        return dct
+
+class PosRange(namedtuple('PRange', ['start', 'stop'])):
```

Comment:
Since we're not using this in Jedi in general and `start <= x <= stop` is really good enough, please remove this.